### ISSUE TITLE
Test of `where -b` block in example does not pass 

### ISSUE THREAD

**Issue by dandavison** [nushell/nushell#7034]
### Describe the bug

The `where` command has the following example:

```
description: "Get all numbers above 3 with an existing block condition",
example: "let a = {$in > 3}; [1, 2, 5, 6] | where -b $a",
``` 

with expected result `[5, 6]`

This should pass but does not.

### How to reproduce

Uncomment the commented-out test in `crates/nu-command/src/filters/where_.rs` (this assumes that https://github.com/nushell/nushell/pull/7035 has been landed) and make this change:

```diff
diff --git a/crates/nu-command/src/example_test.rs b/crates/nu-command/src/example_test.rs
index 018026295..8fcdc329e 100644
--- a/crates/nu-command/src/example_test.rs
+++ b/crates/nu-command/src/example_test.rs
@@ -13,7 +13,7 @@ use crate::To;
 
 #[cfg(test)]
 use super::{
-    Ansi, Date, From, If, Into, LetEnv, Math, Path, Random, Split, SplitColumn, SplitRow, Str,
+    Ansi, Date, From, If, Into, Let, LetEnv, Math, Path, Random, Split, SplitColumn, SplitRow, Str,
     StrJoin, StrLength, StrReplace, Url, Wrap,
 };
 
@@ -47,6 +47,7 @@ pub fn test_examples(cmd: impl Command + 'static) {
         working_set.add_decl(Box::new(Url));
         working_set.add_decl(Box::new(Ansi));
         working_set.add_decl(Box::new(Wrap));
+        working_set.add_decl(Box::new(Let));
         working_set.add_decl(Box::new(LetEnv));
 
         use super::Echo;
```

Then run

```
cargo test -p nu-command filters::where_::test::test_examples
```

You'll see an error like this:

```
input: let a = {$in > 3}; [1, 2, 5, 6] | where -b $a
result: List { vals: [], span: Span { start: 0, end: 0 } }
done: 1.686012ms
thread 'filters::where_::test::test_examples' panicked at
'the example result is different to expected value:
List { vals: [], span: Span { start: 0, end: 0 } }

!=

List { vals: [Int { val: 5, span: Span { start: 0, end: 0} }, Int { val: 6, span: Span { start: 0, end: 0 } }], span: Span { start: 0, end: 0 } }',
crates/nu-command/src/example_test.rs:141:25
```

### Expected behavior

The test should pass, since it works at the REPL:

```
〉let a = {$in > 3}; [1, 2, 5, 6] | where -b $a
╭───╮
│ 5 │
│ 6 │
╰───╯
```

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.70.1                                          |
| branch             | dev                                             |
| commit_hash        | e7c5abf4efa0f5645fc19a3bd777044e8a0f7136        |
| build_os           | macos-x86_64                                    |
| build_target       | x86_64-apple-darwin                             |
| rust_version       | rustc 1.63.0 (4b91a6ea7 2022-08-08)             |
| rust_channel       | 1.63.0-x86_64-apple-darwin                      |
| cargo_version      | cargo 1.63.0 (fd9c4297c 2022-07-01)             |
| pkg_version        | 0.70.1                                          |
| build_time         | 2022-10-20 09:10:05 -04:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by dandavison** [comment]
With `Let` added to the working set there's a different error:

```
thread 'filters::where_::test::test_examples' panicked at 'test eval error in `TODO`: MissingParameter("condition", Span { start: 89, end: 94 })', crates/nu-command/src/example_test.rs:302:25
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: nu_command::example_test::test_examples::eval_block
             at ./src/example_test.rs:302:25
   3: nu_command::example_test::test_examples::eval
             at ./src/example_test.rs:265:9
   4: nu_command::example_test::test_examples::check_example_evaluates_to_expected_output
             at ./src/example_test.rs:218:22
   5: nu_command::example_test::test_examples::test_examples
             at ./src/example_test.rs:47:13
   6: nu_command::example_test::test_examples
             at ./src/example_test.rs:6:5
   7: nu_command::filters::where_::test::test_examples
             at ./src/filters/where_.rs:314:9
   8: nu_command::filters::where_::test::test_examples::{{closure}}
             at ./src/filters/where_.rs:311:5
   9: core::ops::function::FnOnce::call_once
             at /private/tmp/rust-20220812-6672-194b8m/rustc-1.63.0-src/library/core/src/ops/function.rs:248:5
```

**Comment by sholderbach** [comment]
Is this fixed by #7313?

**Comment by sholderbach** [comment]
`where -b` has been removed and supersed by `filter` #7365
